src/main/kotlin/com/github/vol0n/utbotcppclion/actions/GenerateForPredicateAction.kt

```diff
@@ -0,0 +1,246 @@
+            .setMovable(false)
+            .setTitle(title)
+            .setSelectionMode(ListSelectionModel.SINGLE_SELECTION)
+            .setCloseOnEnter(true)
+            .setItemChosenCallback(onChoose)
+            .createPopup()
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+
+        fun sendPredicateToServer(validationType: ValidationType, returnValue: String, predicate: String) {
+            val predicateInfo = buildPredicateInfo(predicate, returnValue, validationType)
+            val predicateRequest = Testgen.PredicateRequest.newBuilder()
+                .setLineRequest(buildLineRequestFromEvent(e))
+                .setPredicateInfo(predicateInfo)
+                .build()
+            if (GeneratorSettingsDialog().showAndGet()) {
+                coroutinesScopeForGrpc.launch {
+                    client.generateForPredicate(predicateRequest).handleTestsResponse()
+                }
+            }
+        }
+
+        fun getReturnValue(validationType: ValidationType, predicate: String) {
+            val textField = ExtendableTextField()
+            textField.minimumSize = Dimension(100, textField.width)
+            textField.text = defaultReturnValues[validationType]
+            textField.selectAll()
+            val popup = JBPopupFactory.getInstance().createComponentPopupBuilder(textField, null)
+                .setFocusable(true)
+                .setRequestFocus(true)
+                .setTitle("Specify Return Value of type ${validationTypeName[validationType]}")
+                .createPopup()
+
+            var canClosePopup = true
+            ComponentValidator(popup).withValidator(Supplier<ValidationInfo?> {
+                val validationResult = returnValueValidators[validationType]?.let { it(textField.text) }
+                if (validationResult == null) {
+                    canClosePopup = true
+                    null
+                } else {
+                    canClosePopup = false
+                    ValidationInfo(validationResult, textField)
+                }
+            }).installOn(textField)
+
+            textField.document.addDocumentListener(object : DocumentAdapter() {
+                override fun textChanged(p0: DocumentEvent) {
+                    ComponentValidator.getInstance(textField).ifPresent { v ->
+                        v.revalidate()
+                    }
+                }
+            })
+
+            textField.addKeyListener(object : KeyAdapter() {
+                override fun keyPressed(e: KeyEvent) {
+                    if (e.keyCode == KeyEvent.VK_ENTER) {
+                        if (canClosePopup) {
+                            popup.cancel()
+                            sendPredicateToServer(validationType, textField.text, predicate)
+                        }
+                    }
+                }
+            })
+            popup.showInBestPositionFor(e.dataContext)
+        }
+
+        fun getPredicate(type: ValidationType) {
```

Comment:
The naming can be improved. It is not just `getPredicate` because it also sends the predicate to server

Reply:
It can return predicate along with getReturnValue and then you can send it to server